pkg/api/component_api.go

```diff
@@ -56,17 +56,60 @@ func (o *DeletePostgresDatabaseOpts) Validate() error {
 	)
 }
 
+// S3Bucket contains the state after an AWS S3
+// bucket has been created
+type S3Bucket struct {
+	ID                     ID
+	Name                   string
+	StackName              string
+	CloudFormationTemplate string
+}
+
+// CreateS3BucketOpts contains the required inputs
+type CreateS3BucketOpts struct {
+	ID        ID
+	Name      string
+	StackName string
+}
+
+// Validate the inputs
+func (o CreateS3BucketOpts) Validate() error {
+	return validation.ValidateStruct(&o,
+		validation.Field(&o.ID, validation.Required),
+		validation.Field(&o.Name, validation.Required),
+		validation.Field(&o.StackName, validation.Required),
+	)
+}
+
+// DeleteS3BucketOpts contains the required inputs
+type DeleteS3BucketOpts struct {
+	ID        ID
+	StackName string
+}
+
+// Validate the inputs
+func (o DeleteS3BucketOpts) Validate() error {
+	return validation.ValidateStruct(&o,
+		validation.Field(&o.ID, validation.Required),
+		validation.Field(&o.StackName, validation.Required),
+	)
+}
+
 // ComponentService defines a set of operations for creating
 // components that integrate with the Kubernetes cluster's
 // applications
 type ComponentService interface {
 	CreatePostgresDatabase(ctx context.Context, opts *CreatePostgresDatabaseOpts) (*PostgresDatabase, error)
 	DeletePostgresDatabase(ctx context.Context, opts *DeletePostgresDatabaseOpts) error
+	CreateS3Bucket(ctx context.Context, opts *CreateS3BucketOpts) (*S3Bucket, error)
```

Comment:
What will we be using the s3 buckets for?

Reply:
Because of the `PermissionsBoundary` we have to create our own lambda function :( buuut, I actually think this might be a good thing long-term, because then we at least have support for creating lambdas from within okctl 🤷🏽 win some, loose some I guess